tests/unittests/tests-checksum/tests-checksum-crc16-ccitt-false.c

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2022 Bennet Blischke <bennet.blischke@haw-hamburg.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+#include <stdint.h>
+
+#include "embUnit/embUnit.h"
+
+#include "checksum/crc16_ccitt.h"
+
+#include "tests-checksum.h"
+
+static int calc_and_compare_crc16_ccitt_false_with_update(const unsigned char *buf,
+                                                          size_t len, size_t split,
+                                                          uint16_t expected)
+{
+    uint16_t result = crc16_ccitt_false_calc(buf, split);
+
+    result = crc16_ccitt_false_update(result, buf + split, len - split);
+    return result == expected;
+}
+
+static int calc_and_compare_crc16_ccitt_false(const unsigned char *buf, size_t len,
+                                              uint16_t expected)
+{
+    uint16_t result = crc16_ccitt_false_calc(buf, len);
+
+    return result == expected;
+}
+
+static void test_checksum_crc16_ccitt_false_sequence_empty(void)
+{
+    unsigned char buf[] = "";
+    uint16_t expect = 0xFFFF;
+
+    TEST_ASSERT(calc_and_compare_crc16_ccitt_false(buf, sizeof(buf) - 1, expect));
+    TEST_ASSERT(calc_and_compare_crc16_ccitt_false_with_update(buf, sizeof(buf) - 1,
+                                                               (sizeof(buf) - 1) / 2, expect));
+}
+
+static void test_checksum_crc16_ccitt_false_sequence_1a(void)
+{
+    unsigned char buf[] = "A";
+    uint16_t expect = 0xB915;
+
+    TEST_ASSERT(calc_and_compare_crc16_ccitt_false(buf, sizeof(buf) - 1, expect));
+    TEST_ASSERT(calc_and_compare_crc16_ccitt_false_with_update(buf, sizeof(buf) - 1,
+                                                               (sizeof(buf) - 1) / 2, expect));
+}
+
+static void test_checksum_crc16_ccitt_false_sequence_256a(void)
+{
+    unsigned char buf[] = "AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA"
+                          "AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA"
+                          "AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA"
+                          "AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA"
+                          "AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA"
+                          "AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA"
+                          "AAAA";
+    uint16_t expect = 0xEA0B;
```

Comment:
Do you have a source for the test vectors? If yes, would be great to cite it here in a comment.

Reply:
I run the test vectors through multiple different implementations of theses crc and checked they matched. E.g. The linux kernel produces the same results for the given vectors. Not sure if it is worth noting in the comments - at the end, it is math. More important it is imo test the API behaves correctly for the edge case empty data.